src/url.rs

```diff
@@ -314,6 +314,45 @@ impl PyMultiHostUrl {
     fn __getnewargs__(&self) -> (String,) {
         (self.__str__(),)
     }
+
+    #[classmethod]
+    #[pyo3(signature=(*, scheme, host, user=None, password=None, port=None, path=None, query=None, fragment=None))]
+    #[allow(clippy::too_many_arguments)]
+    pub fn build(
```

Comment:
this method be on both `PyUrl` (with this signature), and `PyMultiHostUrl` - I guess allowing multiple hosts.

Reply:
Thanks @samuelcolvin - not 100% sure what it is thats needed here, are you saying:

- Implement the method on PyUrl too
- Have PyMultiHostUrl call its ref_url.build for its build function? or implement differently logic for PyMultiHostUrl?